torchvision/prototype/transforms/functional/_color.py

```diff
@@ -318,10 +318,6 @@ def posterize(inpt: features.InputTypeJIT, bits: int) -> features.InputTypeJIT:
 
 
 def solarize_image_tensor(image: torch.Tensor, threshold: float) -> torch.Tensor:
-    bound = 1 if image.is_floating_point() else 255
-    if threshold > bound:
-        raise TypeError(f"Threshold should be less or equal the maximum value of the dtype, but got {threshold}")
-
```

Comment:
@pmeier This check didn't allow me to run the benchmark for float types so I removed it:
```
    return solarize_image_tensor(inpt, threshold=threshold)
  File "./vision/torchvision/prototype/transforms/functional/_color.py", line 323, in solarize_image_tensor
    raise TypeError(f"Threshold should be less or equal the maximum value of the dtype, but got {threshold}")
TypeError: Threshold should be less or equal the maximum value of the dtype, but got 226.6666717529297
```

This is caused by the 255 values hardcoded on the Solarize linspace of AutoAugment. Now that we actually support float32 in all kernels, how do you propose to solve this? I think this should be disconnected from the PR at #6830. We could implement an approach to switch to 1.0 and 255 depending on the type of the input. Later if the aforementioned PR gets merged, we can update with more fine-grained max values that support all integers. Thoughts?